gh-100290: unittest checks file follows pattern

PurityLake opened the pull request:
Added a check in `unittest` module that ensures that any file in a module that is being read, will follow a pattern that is supplied.

My reasoning for putting this test in `loadTestsFromModule` is that the related issue is regarding modules.

<!-- gh-issue-number: gh-100290 -->
* Issue: gh-100290
<!-- /gh-issue-number -->

PurityLake committed:
gh-100290: unittest checks file follows pattern

sobolevn commented on Lib/unittest/loader.py:
```diff
@@ -94,6 +94,10 @@ def loadTestsFromTestCase(self, testCaseClass):
         """Return a suite of all test cases contained in the given module"""
         tests = []
+        if pattern is not None and not fnmatch(
+            os.path.basename(module.__file__), pattern):
```
```suggestion
                os.path.basename(module.__file__), pattern):
```

Extra identation for better readability: body and condition do not visually merge anymore.

sobolevn reviewed (commented):
Please, add a test case for this and a news entry.
Thank you!

PurityLake committed:
Fix formatting in Lib/unittest/loader.py

Suggested improvement to formatting to not make condition blend into the body of the if statement

Co-authored-by: Nikita Sobolev <mail@sobolevn.me>

PurityLake committed:
Added guard condition to loadTestsFromModule if no __file__ is present

PurityLake committed:
pythongh-100290: Added tests for file discovery in unittest

blurb-it[bot] committed:
📜🤖 Added by blurb_it.

PurityLake committed:
gh-100290: Fix bad formatting

PurityLake committed:
Merge branch 'main' into gh-100290-unittest-fix
